sesman/session.c

```diff
@@ -828,12 +779,17 @@ session_start_fork(tbus data, tui8 type, struct SCP_SESSION *s)
             else
             {
                 wait_for_xserver(display);
-                g_snprintf(text, 255, "%d", display);
-                g_setenv("XRDP_SESSVC_DISPLAY", text, 1);
-                g_snprintf(text, 255, ":%d.0", display);
-                g_setenv("DISPLAY", text, 1);
-                /* new style waiting for clients */
-                session_start_sessvc(xpid, wmpid, data, s->username, display);
+                cspid = session_start_chansrv(s->username, display);
+                log_message(LOG_LEVEL_ALWAYS, "waiting for window manager "
+                            "pid(%d) to exit", wmpid);
+                g_waitpid(wmpid);
+                log_message(LOG_LEVEL_ALWAYS, "window manager pid(%d) did "
+                            "exit, cleaning up session", wmpid);
```

Comment:
It would be better to write `window manager (pid %d)` than `window manager pid(%d)` here and two lines above.